modules/registrars/openprovider/Controllers/System/ApiController.php

```diff
@@ -185,15 +185,27 @@ public function updateDnsSecEnabled(array $params)
             'isDnssecEnabled' => (bool)$params['isDnssecEnabled'],
         ];
 
+        $dnssecKeys = [];
         try {
             $domainOp = $this->apiHelper->getDomain($domain);
             $this->apiHelper->updateDomain($domainOp['id'], $args);
+            $updatedDomainOP = $this->apiHelper->getDomain($domain);
```

Comment:
just for the clarification,
can you check these functions are triggering synchronously? may be some caches in the middle

Reply:
It calls to API and gets domain and does not use caches